### PR TITLE
Add CAPI Ubuntu image for DigitalOcean

### PR DESCRIPTION
**Pull request by prksu**
This PR add capi ubuntu image for DigitalOcean provider.

Related https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/140

### CODE REVIEW

File: images/capi/packer/digitalocean/Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	mkdir -p ${HOME}/.packer.d/plugins
+	mkdir -p /tmp/packer-plugins
+	curl -L -o /tmp/packer-plugins/packer-provisioner-goss ${PLUGINS_GOSS_URL}
+	echo ${PLUGINS_GOSS_SHA256-$(BINARYPATHPATTERN)}  */tmp/packer-plugins/packer-provisioner-goss > /tmp/packer-plugins/SHA256SUMS
+	shasum -c /tmp/packer-plugins/SHA256SUMS
+	mv /tmp/packer-plugins/packer-provisioner-goss ${HOME}/.packer.d/plugins
+	chmod +x ${HOME}/.packer.d/plugins/packer-provisioner-goss
+
+.PHONY: plugins
+plugins: ${HOME}/.packer.d/plugins/packer-provisioner-goss
```
Comment (detiber) [inline images/capi/packer/digitalocean/Makefile]: It doesn't look like goss is used anywhere in this provider, this Makefile can likely just be omitted.

Comment (prksu) [reply images/capi/packer/digitalocean/Makefile]: addressed

Comment (prksu) [reply images/capi/packer/digitalocean/Makefile]: thanks

**Comment by prksu** [comment]
@detiber i'll add some docs as soon as possible

**Comment by prksu** [comment]
@detiber i have just pushed docs under `docs/book/src/capi/providers/` please take another look
thanks

File: docs/book/src/capi/capi.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ The Image Builder can be used to build images intended for use with CAPI provide
 * [Azure](./providers/azure.md)
 * Google *TODO*
 * [vSphere](./providers/vsphere.md)
+* [DigitalOcean](./providers/digitalocean.md)
```
Comment (codenrhoden) [inline docs/book/src/capi/capi.md]: nit: Preserve existing list's alphabetical order.

File: docs/book/src/capi/capi.md [hunk 3]
```diff
@@ -26,6 +27,7 @@ Check the Makefile to see a list of images that may be built:
 | `make build-ova-centos-7` |
 | `make build-ova-photon-3` |
 | `make build-ova-ubuntu-1804` |
+| `make build-do-default` |
```
Comment (codenrhoden) [inline docs/book/src/capi/capi.md]: nit: Preserve existing list's alphabetical order.

Comment (prksu) [reply docs/book/src/capi/capi.md]: addressed

File: docs/book/src/capi/capi.md (hunk 2 above)
Comment (prksu) [reply docs/book/src/capi/capi.md]: addressed

**Comment by prksu** [comment]
@codenrhoden yeah, it would be better if sorted alphabetically. i have just updated. thanks

**Comment by codenrhoden** [comment]
/approve

Thanks @prksu !

**Pull request merged**